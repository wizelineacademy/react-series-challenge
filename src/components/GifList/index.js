import React from 'react';
import { Row, Col } from '../Grid';
import Gif from '../Gif';

const GifList = () => {
  return (
    <Row>
      <Col span={4}>
        <Gif />
      </Col>
      <Col span={4}>
        <Gif />
      </Col>
      <Col span={4}>
        <Gif />
      </Col>
    </Row>
  )
}

export default GifList;
