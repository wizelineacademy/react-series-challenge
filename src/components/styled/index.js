import ClearButton from './ClearButton';
import DetailsDiv from './DetailsDiv';
import FavButton from './FavButton';
import Header from './Header';
import ItemDiv from './ItemDiv';
import ItemListDiv from './ItemListDiv';
import SearchButton from './SearchButton';
import SearchDiv from './SearchDiv';
import SearchText from './SearchText';

export {
  ClearButton,
  DetailsDiv,
  FavButton,
  Header,
  ItemDiv,
  ItemListDiv,
  SearchButton,
  SearchDiv,
  SearchText
}
