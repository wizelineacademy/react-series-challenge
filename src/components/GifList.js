import React, { Component } from 'react';
import { Container, Col } from './Grid.styled';
import { GifsRow, GifCard, LikeButton, FullWidthGif, GifTitle } from './GifList.styled'

class GifList extends Component {
  render() {
    const { gifs } = this.props;
    return (
      <Container>
        <GifsRow>
          {(gifs && Object.keys(gifs).length > 0) ?
            Object.keys(gifs).map((key) => {
              const gif = gifs[key];
              const { images } = gif;
              const { fixed_width: image } = images;
              return (
                <Col size="3" key={gif.id}>
                  <GifCard>
                    <LikeButton
                      liked={gif.liked}
                      onClick={() => this.props.onLikeClick(gif)}
                    >
                      <i className="fas fa-heart" />
                    </LikeButton>
                    <FullWidthGif src={image.url} />
                    <GifTitle><span>{gif.title}</span></GifTitle>
                  </GifCard>
                </Col>
              )
            })
            :
            <h4>Not Found Gifs</h4>
          }
        </GifsRow>
      </Container>
    );
  }
}

export default GifList;