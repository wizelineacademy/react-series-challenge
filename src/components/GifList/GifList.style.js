import styled from 'styled-components'

export const GifWrapper = styled.div`
  width: 100%;
  height: 100%;
`

export const GifStyledList = styled.ul`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  height: 2100px;
  width: 1000px;
  padding: 5px;
  margin: auto;
`

export const GifItemWrapper = styled.div`
  margin: 5px;
  width: fit-content;
  height: fit-content;
  cursor: pointer;

  .favorite {
    visibility: hidden;
  }

  :hover {
    .favorite {
      visibility: visible;
    }
  }
`
export const FavoriteButton = styled.img`
  position: absolute;
  width: 25px;
  height: 25px;
  cursor: pointer;
  margin: 5px 0 0 5px;
`
GifStyledList.displayName = 'GifStyledList'
