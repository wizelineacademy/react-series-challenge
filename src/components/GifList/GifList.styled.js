import styled from 'styled-components';

export const MasonryContainer = styled.div`
  display: flex;
  flex-flow: row wrap;
  margin-left: -8px;
  width: 100%;
  margin-top: 15px;
`;