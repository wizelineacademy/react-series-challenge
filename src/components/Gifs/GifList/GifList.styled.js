import styled from 'styled-components'

const GifListWrapper = styled.section`
  display: grid;
  grid-template-columns: repeat(4, 248px);
  grid-gap: 1rem;
`

export {
  GifListWrapper
}