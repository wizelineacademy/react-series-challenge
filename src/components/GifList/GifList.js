import React from 'react';
import {bindActionCreators} from "redux";
import {connect} from 'react-redux';
import {Gif} from "../Gif/Gif";
import {MasonryContainer} from "./GifList.styled";
import favoritesGifs from "../../actions/favoritesGifs";


const isFavorited = (favorites, gifId) => {
  return favorites[gifId] !== undefined;
}

export const GifList = ({ gifs, addGif, removeGif, favoritesGifs }) => {
  return (
    <MasonryContainer>
      { gifs.map(gif =>
        <Gif
          addGif={addGif}
          removeGif={removeGif}
          key={gif.id}
          gif={gif}
          isFavorited={isFavorited(favoritesGifs.allGifs, gif.id)}
        />
      )}
    </MasonryContainer>
  )
};
const mapStateToProps = (state) => {
  const { favoritesGifs } = state;
  return { favoritesGifs };
};
const mapDispatchToProps = (dispatch) => {
  const { addGif, removeGif } = favoritesGifs.creators;

  return bindActionCreators({
    addGif,
    removeGif
  }, dispatch);
};
export default connect(mapStateToProps, mapDispatchToProps)(GifList);

