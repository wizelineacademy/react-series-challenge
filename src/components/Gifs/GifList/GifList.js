import React from 'react';
import GifCard from '../GifCard/GifCard'
import { GifListWrapper } from './GifList.styled'

const GifList = ({data}) => {
  let gifs

  if (data) {
    gifs = data.map((gif, index) => {
      const {id, images, title} = gif
      const dataGif = {
        id,
        images,
        title
      }

      return (
        <GifCard
          gif={dataGif}
          key={dataGif.id}
          // addFavorites={addFavorites}
        />
      )
    })
  } else {
    gifs = 'No gifs to Show'
  }

  return (
    <GifListWrapper>
      {gifs}
    </GifListWrapper>
  )
}

export default GifList