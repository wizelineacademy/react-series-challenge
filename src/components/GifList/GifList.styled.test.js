import React from 'react';
import {MasonryContainer} from './GifList.styled';
import renderer from 'react-test-renderer';

describe('Gif list styled components', () => {
  it('masonry container renders correctly', () => {
    const tree = renderer
      .create(<MasonryContainer>Child</MasonryContainer>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
