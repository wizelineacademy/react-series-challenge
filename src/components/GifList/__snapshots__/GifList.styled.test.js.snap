// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gif list styled components masonry container renders correctly 1`] = `
<div
  className="sc-bdVaJa kPXkrG"
>
  Child
</div>
`;
